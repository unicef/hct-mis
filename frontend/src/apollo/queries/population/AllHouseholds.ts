import { gql } from 'apollo-boost';

export const AllHouseholds = gql`
  query AllHouseholds(
    $after: String
    $before: String
    $first: Int
    $last: Int
    $businessArea: String
    $orderBy: String
    $familySize: String
    $programs: [ID]
    $headOfHouseholdFullNameIcontains: String
    $adminArea: ID
    $search: String
    $residenceStatus: String
    $lastRegistrationDate: String
    $admin2: [ID]
    $withdrawn: Boolean
  ) {
    allHouseholds(
      after: $after
      before: $before
      first: $first
      last: $last
      businessArea: $businessArea
      size: $familySize
      orderBy: $orderBy
      programs: $programs
      headOfHousehold_FullName_Startswith: $headOfHouseholdFullNameIcontains
      adminArea: $adminArea
      search: $search
      residenceStatus: $residenceStatus
      lastRegistrationDate: $lastRegistrationDate
      admin2: $admin2
      withdrawn: $withdrawn
    ) {
      pageInfo {
        hasNextPage
        hasPreviousPage
        startCursor
        endCursor
      }
      totalCount
      edges {
        cursor
        node {
          id
          status
          unicefId
          hasDuplicates
          sanctionListPossibleMatch
          sanctionListConfirmedMatch
          headOfHousehold {
            id
            fullName
          }
          size
          admin2 {
            id
            name
          }
          residenceStatus
          totalCashReceived
          currency
          lastRegistrationDate
          oldPrograms {
            edges {
              node {
                id
                name
              }
            }
          }
        }
      }
    }
  }
`;
