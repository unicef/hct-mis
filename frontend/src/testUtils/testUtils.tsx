import React, { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import { TestProviders } from './testProviders';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(<BrowserRouter>{ui}</BrowserRouter>, {
    wrapper: TestProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export const makeApolloLoadingMock = (mocks) => {
  let mocksWithDelay = cloneDeep(mocks);
  for (const mock of mocksWithDelay) {
    mock.request.delay = 1e21;
  }
  return mocksWithDelay;
};
