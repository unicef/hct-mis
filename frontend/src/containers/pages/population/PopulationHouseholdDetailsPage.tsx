import { Box, Grid, Typography } from '@material-ui/core';
import Paper from '@material-ui/core/Paper';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import styled from 'styled-components';
import { BreadCrumbsItem } from '../../../components/core/BreadCrumbs';
import { FlagTooltip } from '../../../components/core/FlagTooltip';
import { WarningTooltip } from '../../../components/core/WarningTooltip';
import { LabelizedField } from '../../../components/core/LabelizedField';
import { LoadingComponent } from '../../../components/core/LoadingComponent';
import { PageHeader } from '../../../components/core/PageHeader';
import { PermissionDenied } from '../../../components/core/PermissionDenied';
import { HouseholdVulnerabilities } from '../../../components/population/HouseholdVulnerabilities/HouseholdVulnerabilities';
import { UniversalMoment } from '../../../components/core/UniversalMoment';
import { hasPermissions, PERMISSIONS } from '../../../config/permissions';
import { useBusinessArea } from '../../../hooks/useBusinessArea';
import { usePermissions } from '../../../hooks/usePermissions';
import {
  isPermissionDeniedError,
  renderSomethingOrDash,
} from '../../../utils/utils';
import {
  HouseholdNode,
  useAllHouseholdsFlexFieldsAttributesQuery,
  useGrievancesChoiceDataQuery,
  useHouseholdChoiceDataQuery,
  useHouseholdQuery,
} from '../../../__generated__/graphql';
import { HouseholdCompositionTable } from '../../tables/population/HouseholdCompositionTable/HouseholdCompositionTable';
import { HouseholdIndividualsTable } from '../../tables/population/HouseholdIndividualsTable/HouseholdIndividualsTable';
import { PaymentRecordHouseholdTable } from '../../tables/payments/PaymentRecordHouseholdTable';
import { UniversalActivityLogTable } from '../../tables/UniversalActivityLogTable';
import { HouseholdDetails } from '../../../components/population/HouseholdDetails';
import { Title } from '../../../components/core/Title';

const Container = styled.div`
  padding: 20px;
  && {
    display: flex;
    flex-direction: column;
    width: 100%;
  }
`;

const Overview = styled(Paper)`
  padding: ${({ theme }) => theme.spacing(8)}px
    ${({ theme }) => theme.spacing(11)}px;
  margin-top: 20px;
  &:first-child {
    margin-top: 0;
  }
`;

const SubTitle = styled(Typography)`
  && {
    font-size: 16px;
  }
`;

export const PopulationHouseholdDetailsPage = (): React.ReactElement => {
  const { t } = useTranslation();
  const { id } = useParams();
  const businessArea = useBusinessArea();
  const permissions = usePermissions();

  const { data, loading, error } = useHouseholdQuery({
    variables: { id },
    fetchPolicy: 'cache-and-network',
  });
  const {
    data: flexFieldsData,
    loading: flexFieldsDataLoading,
  } = useAllHouseholdsFlexFieldsAttributesQuery();
  const {
    data: choicesData,
    loading: choicesLoading,
  } = useHouseholdChoiceDataQuery();
  const {
    data: grievancesChoices,
    loading: grievancesChoicesLoading,
  } = useGrievancesChoiceDataQuery();

  if (
    loading ||
    choicesLoading ||
    flexFieldsDataLoading ||
    grievancesChoicesLoading
  )
    return <LoadingComponent />;

  if (isPermissionDeniedError(error)) return <PermissionDenied />;

  if (
    !data ||
    !choicesData ||
    !grievancesChoices ||
    !flexFieldsData ||
    permissions === null
  )
    return null;

  const breadCrumbsItems: BreadCrumbsItem[] = [
    {
      title: t('Households'),
      to: `/${businessArea}/population/household`,
    },
  ];

  const { household } = data;

  return (
    <>
      <PageHeader
        title={`${t('Household ID')}: ${renderSomethingOrDash(
          household?.unicefId,
        )}`}
        breadCrumbs={
          hasPermissions(
            PERMISSIONS.POPULATION_VIEW_HOUSEHOLDS_LIST,
            permissions,
          )
            ? breadCrumbsItems
            : null
        }
        flags={
          <>
            <Box mr={2}>
              {household?.hasDuplicates && (
                <WarningTooltip
                  confirmed
                  message={t('Houesehold has Duplicates')}
                />
              )}
            </Box>
            <Box mr={2}>
              {household?.sanctionListPossibleMatch && (
                <FlagTooltip message={t('Sanction List Possible Match')} />
              )}
            </Box>
            <Box mr={2}>
              {household?.sanctionListConfirmedMatch && (
                <FlagTooltip
                  message={t('Sanction List Confirmed Match')}
                  confirmed
                />
              )}
            </Box>
          </>
        }
      />
      <HouseholdDetails
        choicesData={choicesData}
        household={household as HouseholdNode}
        businessArea={businessArea}
        grievancesChoices={grievancesChoices}
      />
      <HouseholdCompositionTable household={household as HouseholdNode} />
      <Container>
        {household?.individuals?.edges?.length ? (
          <HouseholdIndividualsTable
            choicesData={choicesData}
            household={household as HouseholdNode}
          />
        ) : null}
        {hasPermissions(
          PERMISSIONS.PRORGRAMME_VIEW_LIST_AND_DETAILS,
          permissions,
        ) && (
          <PaymentRecordHouseholdTable
            openInNewTab
            household={household as HouseholdNode}
            businessArea={businessArea}
            canViewPaymentRecordDetails={hasPermissions(
              PERMISSIONS.PROGRAMME_VIEW_PAYMENT_RECORD_DETAILS,
              permissions,
            )}
          />
        )}
        <HouseholdVulnerabilities
          household={household as HouseholdNode}
          flexFieldsData={flexFieldsData}
        />
        <Overview>
          <Title>
            <Typography variant='h6'>{t('Registration Details')}</Typography>
          </Title>
          <Grid container spacing={6}>
            <Grid item xs={3}>
              <LabelizedField label={t('Source')}>
                <div>{household?.registrationDataImport?.dataSource}</div>
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Import name')}>
                <div>{household?.registrationDataImport?.name}</div>
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('Registration Date')}>
                <div>
                  <UniversalMoment>
                    {household?.lastRegistrationDate}
                  </UniversalMoment>
                </div>
              </LabelizedField>
            </Grid>
            <Grid item xs={3}>
              <LabelizedField label={t('User name')}>
                {household?.registrationDataImport?.importedBy?.email}
              </LabelizedField>
            </Grid>
          </Grid>
          {household?.registrationDataImport?.dataSource === 'XLS' ? null : (
            <>
              <hr />
              <SubTitle variant='h6'>{t('Data Collection')}</SubTitle>
              <Grid container spacing={6}>
                <Grid item xs={3}>
                  <LabelizedField label={t('Start time')}>
                    <UniversalMoment>{household?.start}</UniversalMoment>
                  </LabelizedField>
                </Grid>
                <Grid item xs={3}>
                  <LabelizedField label={t('End time')}>
                    <UniversalMoment>
                      {household?.firstRegistrationDate}
                    </UniversalMoment>
                  </LabelizedField>
                </Grid>
                <Grid item xs={3}>
                  <LabelizedField label={t('Device ID')}>
                    {household?.deviceid}
                  </LabelizedField>
                </Grid>
              </Grid>
            </>
          )}
        </Overview>
      </Container>
      {hasPermissions(PERMISSIONS.ACTIVITY_LOG_VIEW, permissions) && (
        <UniversalActivityLogTable objectId={data.household?.id} />
      )}
    </>
  );
};
