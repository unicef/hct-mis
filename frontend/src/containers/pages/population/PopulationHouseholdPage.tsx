import { Box } from '@material-ui/core';
import get from 'lodash/get';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';
import {
  ProgramNode,
  useAllProgramsForChoicesQuery,
  useHouseholdChoiceDataQuery,
} from '../../../__generated__/graphql';
import { LoadingComponent } from '../../../components/core/LoadingComponent';
import { PageHeader } from '../../../components/core/PageHeader';
import { PermissionDenied } from '../../../components/core/PermissionDenied';
import { HouseholdFilters } from '../../../components/population/HouseholdFilter';
import { hasPermissions, PERMISSIONS } from '../../../config/permissions';
import { useBusinessArea } from '../../../hooks/useBusinessArea';
import { usePermissions } from '../../../hooks/usePermissions';
import { getFilterFromQueryParams } from '../../../utils/utils';
import { HouseholdTable } from '../../tables/population/HouseholdTable';

const initialFilter = {
  search: '',
  program: '',
  residenceStatus: '',
  admin2: '',
  householdSizeMin: '',
  householdSizeMax: '',
  orderBy: 'unicef_id',
  withdrawn: null,
};

export const PopulationHouseholdPage = (): React.ReactElement => {
  const { t } = useTranslation();
  const location = useLocation();

  const [filter, setFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );
  const [appliedFilter, setAppliedFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );

  const businessArea = useBusinessArea();
  const permissions = usePermissions();

  const { data, loading } = useAllProgramsForChoicesQuery({
    variables: { businessArea },
    fetchPolicy: 'cache-and-network',
  });
  const {
    data: choicesData,
    loading: choicesLoading,
  } = useHouseholdChoiceDataQuery({
    variables: { businessArea },
    fetchPolicy: 'cache-first',
  });

  if (loading || choicesLoading) return <LoadingComponent />;
  if (permissions === null) return null;

  if (!hasPermissions(PERMISSIONS.POPULATION_VIEW_HOUSEHOLDS_LIST, permissions))
    return <PermissionDenied />;

  if (!choicesData) return null;

  const allPrograms = get(data, 'allPrograms.edges', []);
  const programs = allPrograms.map((edge) => edge.node);

  return (
    <>
      <PageHeader title={t('Households')} />
      <HouseholdFilters
        programs={programs as ProgramNode[]}
        filter={filter}
        choicesData={choicesData}
        setFilter={setFilter}
        initialFilter={initialFilter}
        appliedFilter={appliedFilter}
        setAppliedFilter={setAppliedFilter}
      />
      <Box
        display='flex'
        flexDirection='column'
        data-cy='page-details-container'
      >
        <HouseholdTable
          filter={appliedFilter}
          businessArea={businessArea}
          choicesData={choicesData}
          canViewDetails={hasPermissions(
            PERMISSIONS.POPULATION_VIEW_HOUSEHOLDS_DETAILS,
            permissions,
          )}
        />
      </Box>
    </>
  );
};
