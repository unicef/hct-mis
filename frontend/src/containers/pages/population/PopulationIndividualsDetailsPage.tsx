import { Box } from '@mui/material';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import styled from 'styled-components';
import {
  IndividualNode,
  useAllIndividualsFlexFieldsAttributesQuery,
  useGrievancesChoiceDataQuery,
  useHouseholdChoiceDataQuery,
  useIndividualQuery,
} from '@generated/graphql';
import { BreadCrumbsItem } from '@components/core/BreadCrumbs';
import { LoadingComponent } from '@components/core/LoadingComponent';
import { PageHeader } from '@components/core/PageHeader';
import { PermissionDenied } from '@components/core/PermissionDenied';
import { IndividualBioData } from '@components/population/IndividualBioData/IndividualBioData';
import { IndividualFlags } from '@components/population/IndividualFlags';
import { IndividualPhotoModal } from '@components/population/IndividualPhotoModal';
import { IndividualVulnerabilities } from '@components/population/IndividualVulnerabilities/IndividualVunerabilities';
import { hasPermissions, PERMISSIONS } from '../../../config/permissions';
import { useBaseUrl } from '@hooks/useBaseUrl';
import { usePermissions } from '@hooks/usePermissions';
import { isPermissionDeniedError } from '@utils/utils';
import { UniversalActivityLogTable } from '../../tables/UniversalActivityLogTable';
import { GenericAdminButton } from "@core/AdminButton";

const Container = styled.div`
  padding: 20px;
  && {
    display: flex;
    flex-direction: column;
    width: 100%;
  }
`;

export function PopulationIndividualsDetailsPage(): React.ReactElement {
  const { t } = useTranslation();
  const { id } = useParams();
  const { baseUrl, businessArea } = useBaseUrl();
  const permissions = usePermissions();

  const { data, loading, error } = useIndividualQuery({
    variables: {
      id,
    },
    fetchPolicy: 'cache-and-network',
  });

  const { data: choicesData, loading: choicesLoading } =
    useHouseholdChoiceDataQuery();

  const { data: flexFieldsData, loading: flexFieldsDataLoading } =
    useAllIndividualsFlexFieldsAttributesQuery();

  const { data: grievancesChoices, loading: grievancesChoicesLoading } =
    useGrievancesChoiceDataQuery();

  if (
    loading ||
    choicesLoading ||
    flexFieldsDataLoading ||
    grievancesChoicesLoading
  )
    return <LoadingComponent />;

  if (isPermissionDeniedError(error)) return <PermissionDenied />;

  if (
    !data ||
    !choicesData ||
    !flexFieldsData ||
    !grievancesChoices ||
    permissions === null
  )
    return null;

  const breadCrumbsItems: BreadCrumbsItem[] = [
    {
      title: 'Individuals',
      to: `/${baseUrl}/population/individuals`,
    },
  ];

  const { individual } = data;

  return (
    <>
      <PageHeader
        title={`${t('Individual ID')}: ${individual?.unicefId}`}
        breadCrumbs={
          hasPermissions(
            PERMISSIONS.POPULATION_VIEW_INDIVIDUALS_LIST,
            permissions,
          )
            ? breadCrumbsItems
            : null
        }
        flags={
          <>
            <IndividualFlags individual={individual} />
            <GenericAdminButton currentUrl={document.location.href} />
          </>
        }
      >
        <Box mr={2}>
          {individual?.photo ? (
            <IndividualPhotoModal individual={individual as IndividualNode} />
          ) : null}
        </Box>
      </PageHeader>

      <Container>
        <IndividualBioData
          baseUrl={baseUrl}
          businessArea={businessArea}
          individual={individual as IndividualNode}
          choicesData={choicesData}
          grievancesChoices={grievancesChoices}
        />
        <IndividualVulnerabilities
          flexFieldsData={flexFieldsData}
          individual={individual as IndividualNode}
        />
        {hasPermissions(PERMISSIONS.ACTIVITY_LOG_VIEW, permissions) && (
          <UniversalActivityLogTable objectId={individual?.id} />
        )}
      </Container>
    </>
  );
}
