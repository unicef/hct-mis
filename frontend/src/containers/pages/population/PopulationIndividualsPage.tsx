import { Box } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';
import {
  useHouseholdChoiceDataQuery,
  useIndividualChoiceDataQuery,
} from '@generated/graphql';
import { LoadingComponent } from '@components/core/LoadingComponent';
import { PageHeader } from '@components/core/PageHeader';
import { PermissionDenied } from '@components/core/PermissionDenied';
import { IndividualsFilter } from '@components/population/IndividualsFilter';
import { hasPermissions, PERMISSIONS } from '../../../config/permissions';
import { useBaseUrl } from '@hooks/useBaseUrl';
import { usePermissions } from '@hooks/usePermissions';
import { getFilterFromQueryParams } from '@utils/utils';
import { IndividualsListTable } from '../../tables/population/IndividualsListTable';

export function PopulationIndividualsPage(): React.ReactElement {
  const { t } = useTranslation();
  const location = useLocation();
  const { businessArea } = useBaseUrl();
  const permissions = usePermissions();
  const { data: householdChoicesData, loading: householdChoicesLoading } =
    useHouseholdChoiceDataQuery();

  const { data: individualChoicesData, loading: individualChoicesLoading } =
    useIndividualChoiceDataQuery();

  const initialFilter = {
    search: '',
    documentType: individualChoicesData?.documentTypeChoices?.[0]?.value,
    documentNumber: '',
    admin2: '',
    sex: '',
    ageMin: '',
    ageMax: '',
    flags: [],
    orderBy: 'unicef_id',
    status: '',
    lastRegistrationDateMin: '',
    lastRegistrationDateMax: '',
  };


  const [filter, setFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );
  const [appliedFilter, setAppliedFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );

  if (householdChoicesLoading || individualChoicesLoading)
    return <LoadingComponent />;

  if (!individualChoicesData || !householdChoicesData || permissions === null)
    return null;

  if (
    !hasPermissions(PERMISSIONS.POPULATION_VIEW_INDIVIDUALS_LIST, permissions)
  )
    return <PermissionDenied />;

  return (
    <>
      <PageHeader title={t('Individuals')} />
      <IndividualsFilter
        filter={filter}
        choicesData={individualChoicesData}
        setFilter={setFilter}
        initialFilter={initialFilter}
        appliedFilter={appliedFilter}
        setAppliedFilter={setAppliedFilter}
      />
      <Box
        display="flex"
        flexDirection="column"
        data-cy="page-details-container"
      >
        <IndividualsListTable
          filter={appliedFilter}
          businessArea={businessArea}
          choicesData={householdChoicesData}
          canViewDetails={hasPermissions(
            PERMISSIONS.POPULATION_VIEW_INDIVIDUALS_DETAILS,
            permissions,
          )}
        />
      </Box>
    </>
  );
}
