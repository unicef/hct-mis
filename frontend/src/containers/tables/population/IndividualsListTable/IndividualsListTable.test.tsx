import { MockedProvider } from '@apollo/react-testing';
import React from 'react';
import { act } from '@testing-library/react';
import wait from 'waait';
import { IndividualsListTable } from '.';
import { ApolloLoadingLink, render } from '../../../../testUtils/testUtils';
import { fakeHouseholdChoices } from '../../../../../fixtures/population/fakeHouseholdChoices';
import { fakeApolloAllIndividualsForPopulationTable } from '../../../../../fixtures/population/fakeApolloAllIndividualsForPopulationTable';

describe('containers/tables/population/IndividualsListTable', () => {
  const initialFilter = {
    search: '',
    searchType: 'individual_id',
    admin2: '',
    sex: '',
    ageMin: '',
    ageMax: '',
    flags: [],
    orderBy: 'unicef_id',
    status: '',
    lastRegistrationDateMin: '',
    lastRegistrationDateMax: '',
  };

  it('should render with data', async () => {
    const { container } = render(
      <MockedProvider
        addTypename={false}
        mocks={fakeApolloAllIndividualsForPopulationTable}
      >
        <IndividualsListTable
          businessArea='afghanistan'
          filter={initialFilter}
          canViewDetails={true}
          choicesData={fakeHouseholdChoices}
        />
      </MockedProvider>,
    );
    await act(() => wait(0)); // wait for response

    expect(container).toMatchSnapshot();
  });

  it('should render loading', () => {
    const { container } = render(
      <MockedProvider
        link={new ApolloLoadingLink()}
        addTypename={false}
        mocks={fakeApolloAllIndividualsForPopulationTable}
      >
        <IndividualsListTable
          businessArea='afghanistan'
          filter={initialFilter}
          canViewDetails={true}
          choicesData={fakeHouseholdChoices}
        />
      </MockedProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
