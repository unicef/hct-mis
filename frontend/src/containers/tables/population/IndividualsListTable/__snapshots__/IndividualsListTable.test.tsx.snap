// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/tables/population/IndividualsListTable should render loading 1`] = `
<div>
  <div
    class="sc-bdVaJa ksQzqc"
  >
    <div
      class="makeStyles-root-13"
    >
      <div
        class="MuiPaper-root makeStyles-paper-14 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiTableContainer-root"
        >
          <div
            class="MuiBox-root MuiBox-root-19"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular makeStyles-root-20 MuiToolbar-gutters"
            >
              <h6
                class="MuiTypography-root makeStyles-title-22 MuiTypography-h6"
                data-cy="table-title"
              >
                Individuals
              </h6>
            </div>
            <div
              class="MuiBox-root MuiBox-root-23"
            />
          </div>
          <table
            aria-label="enhanced table"
            aria-labelledby="tableTitle"
            class="MuiTable-root makeStyles-table-15"
            data-cy="table-title"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  data-cy="sanction-list-possible-match"
                  scope="col"
                >
                  <span
                    class="sc-bZQynM fgJQBP"
                    data-cy="table-label"
                  />
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  data-cy="individual-id"
                  scope="col"
                >
                  <span
                    class="sc-bZQynM fgJQBP"
                    data-cy="table-label"
                  >
                    Individual ID
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  data-cy="individual-name"
                  scope="col"
                >
                  <span
                    class="sc-bZQynM fgJQBP"
                    data-cy="table-label"
                  >
                    Individual
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  data-cy="household-id"
                  scope="col"
                >
                  <span
                    class="sc-bZQynM fgJQBP"
                    data-cy="table-label"
                  >
                    Household ID
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  data-cy="relationship"
                  scope="col"
                >
                  <span
                    class="sc-bZQynM fgJQBP"
                    data-cy="table-label"
                  >
                    Relationship to HoH
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  data-cy="individual-age"
                  scope="col"
                >
                  <span
                    class="sc-bZQynM fgJQBP"
                    data-cy="table-label"
                  >
                    Age
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  data-cy="individual-sex"
                  scope="col"
                >
                  <span
                    class="sc-bZQynM fgJQBP"
                    data-cy="table-label"
                  >
                    Gender
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  data-cy="individual-location"
                  scope="col"
                >
                  <span
                    class="sc-bZQynM fgJQBP"
                    data-cy="table-label"
                  >
                    Administrative Level 2
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root"
                data-cy="table-row"
                style="height: 70px; min-height: 70px;"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                  colspan="8"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root"
          data-cy="table-pagination"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
          >
            <div
              class="MuiTablePagination-spacer"
            />
            <p
              class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
              id="mui-5806"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="mui-5806 mui-73230"
                class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                id="mui-73230"
                role="button"
                tabindex="0"
              >
                10
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            >
              1-NaN of undefined
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Previous page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                title="Previous page"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Next page"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                title="Next page"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`containers/tables/population/IndividualsListTable should render with data 1`] = `
<div>
  <div
    class="sc-bdVaJa ksQzqc"
  >
    <div>
      Unexpected error
    </div>
  </div>
</div>
`;
